mmdet/models/detectors/base.py

```diff
@@ -213,59 +211,29 @@ def forward(self, img, img_metas, return_loss=True, **kwargs):
         else:
             return self.forward_test(img, img_metas, **kwargs)
 
-    def forward_export(self, imgs, img_metas=None):
-        """
-        This function is called from the function `forward_test` in one of the cases:
-        * During export of the model
-        * During compression of the model by NNCF, for building graph of
-          the model using `dummy_forward` function. In these cases
-          * `forward_export_context` is used,
-          * `img_metas` is passed to the model using the field `self.img_metas`
-        * During compression of the model by NNCF, for initialization of
-          the model. In this case
-          * `forward_nncf_initialization_context` is used
-          * `img_metas` is passed to the model through parameter of this function.
-
-        See the methods `forward_export_context` and `forward_nncf_initialization_context` below.
-        """
-
-        if img_metas is None and self.img_metas is None:
-            raise RuntimeError("forward_export is called with zero img_metas.")
-
-        if img_metas is not None and self.img_metas is not None:
-            raise RuntimeError('forward_export is called with non-zero img_metas inside forward_export_context.')
-
+    def forward_export(self, imgs):
         from torch.onnx.operators import shape_as_tensor
-        if img_metas is None:
-            img_metas = self.img_metas
+        assert self.img_metas, 'Error: forward_export should be called inside forward_export_context'
 
         img_shape = shape_as_tensor(imgs[0])
         imgs_per_gpu = int(imgs[0].size(0))
         assert imgs_per_gpu == 1
-        assert len(img_metas[0]) == imgs_per_gpu, f'img_metas={img_metas}'
-        img_metas[0][0]['img_shape'] = img_shape[2:4]
+        assert len(self.img_metas[0]) == imgs_per_gpu, f'self.img_metas={self.img_metas}'
+        self.img_metas[0][0]['img_shape'] = img_shape[2:4]
 
-        retval = self.simple_test(imgs[0], img_metas[0], postprocess=False)
-        return retval
+        return self.simple_test(imgs[0], self.img_metas[0], postprocess=False)
 
     @contextmanager
     def forward_export_context(self, img_metas):
-        """
```

Comment:
Why do you remove docsting?

Reply:
It was revert changes of https://github.com/openvinotoolkit/mmdetection/pull/136.
https://github.com/openvinotoolkit/mmdetection/pull/168 not actual for dp/nous_sdk_api